diplomacy/src/Nodes.scala

```diff
@@ -75,7 +95,8 @@ abstract class NodeImp[D, U, EO, EI, B <: Data]
   extends Object with InwardNodeImp[D, U, EI, B] with OutwardNodeImp[D, U, EO, B]
 
 /** [[SimpleNodeImp]] contains Master and Slave Interface implementation, edge of which are same.
-  *
+  * If only one direction of transaction is permitted, we can always use the [[SimpleNodeImp]]
+  * For example, IntImp is an interrupt transfer protocol in which all signals shall flow from a master to a slave. There is no backwards handshake flow from a slave to master.
```

Comment:
I think this has no relationship to `SimpleNodeImp`, since almost all protocol implementation are extends from `SimpleNodeImp`.
in the `MixedAdapterNode`, it consumes two `NodeImp`, `inner` as `InwardNodeImp`, while `outer` as `OutwardNodeImp`, `inner` can be the `TLImp` and `outer` can be `AXI4Imp`, both of these two implementations are `SimpleNodeImp`, but `MixedAdapterNode` only use inward side or outward side of them.
Thus I don't think `one direction of transaction is permitted` is necessary.
What do you think?

Reply:
oh yes, it is not accurate to say `permitted` here. I think it is better to say `when only one direction of transaction is considered or when the edge parameters are the same for both inner and oter side`

The reason why `AXI4Imp` is a `SimpleNodeImp` is because rocket-chip does not fully implement `AXI4` protocol. The `AXI4EdgeParameters` only tells how to generate bundle via edge connections, which is the same for both `EO` and `EI`. 

For the `TLImp`, I think it is a `NodeImp`, not a `Simple` one.